GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -200,6 +201,46 @@ public IBuildServerCredentials GetBuildServerCredentials(IBuildServerAdapter bui
             }
         }
 
+        /// <summary>
+        /// Get a "repo shortname" from the current repo url
+        /// There is no official Git repo shortname, this is one possible definition, extracting the filename without extension for the remote repo
+        /// This function couild have been included in GitModule
+        /// </summary>
+        /// <returns>URL filename part if a remote exists and empty or null if no repo URL</returns>
+        private string GetRepoShortname()
+        {
+            //Extract "name of repo" from remote url
+            string remoteName = Module.GetCurrentRemote();
+            if (remoteName.IsNullOrWhiteSpace())
+            {
+                //No remote for the branch, for instance a submodule. Use first remote.
+                var remotes = Module.GetRemotes();
+                if (remotes.Length > 0)
+                {
+                    remoteName = remotes[0];
+                }
+            }
+            var remoteUrl = Module.GetSetting(string.Format(SettingKeyString.RemoteUrl, remoteName));
+            var repoName = Path.GetFileNameWithoutExtension(remoteUrl);
+
+            return repoName;
+        }
+
+        /// <summary>
+        /// Replace variables for the project string with the current "repo shortname"
+        /// </summary>
+        /// <param name="projectNames">build server specific format, compatible with the variable format (Windows style %NAME%)</param>
+        /// <returns>projectNames with variables replaced</returns>
+        public string ReplaceVariables(string projectNames)
+        {
+            var repoName = GetRepoShortname();
+
+            if (repoName.IsNullOrWhiteSpace())
+                return projectNames;
+
+            return projectNames.Replace("%REPO_SHORTNAME%", repoName);
```

Comment:
Can we align the new variables with what we have for scripts?
https://github.com/gitextensions/gitextensions/blob/d6fdd5b1c8b9779604c0dc506d1fc5f71c3aa2fb/GitUI/Script/ScriptRunner.cs#L433-L463

JIRA plugin also has a bunch of veriables (coming from Atlassian.Jira.Issue type):
![image](https://user-images.githubusercontent.com/4403806/35182530-eda136b4-fe2a-11e7-97d8-0b8c95452bdb.png)

Reply:
What I meant was - instead of using "%VAR_NAME%" notation use notation "{VarName}" which is used in other places

Reply:
> I suggest renaming %REPO_SHORTNAME% to {cRepoShortName}